src/test/java/uk.gov.ons.census.action/messaging/UndeliveredMailReceiverTest.java

```diff
@@ -0,0 +1,277 @@
+package uk.gov.ons.census.action.messaging;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import uk.gov.ons.census.action.builders.FieldworkFollowupBuilder;
+import uk.gov.ons.census.action.model.dto.FieldworkFollowup;
+import uk.gov.ons.census.action.model.dto.FulfilmentInformation;
+import uk.gov.ons.census.action.model.dto.Payload;
+import uk.gov.ons.census.action.model.dto.ResponseManagementEvent;
+import uk.gov.ons.census.action.model.entity.Case;
+import uk.gov.ons.census.action.model.entity.UacQidLink;
+import uk.gov.ons.census.action.model.repository.CaseRepository;
+import uk.gov.ons.census.action.model.repository.UacQidLinkRepository;
+
+public class UndeliveredMailReceiverTest {
+  @Test
+  public void testHappyPathWithQid() {
+    RabbitTemplate rabbitTemplate = mock(RabbitTemplate.class);
+    CaseRepository caseRepository = mock(CaseRepository.class);
+    UacQidLinkRepository uacQidLinkRepository = mock(UacQidLinkRepository.class);
+    FieldworkFollowupBuilder fieldworkFollowupBuilder = mock(FieldworkFollowupBuilder.class);
+
+    UndeliveredMailReceiver underTest =
+        new UndeliveredMailReceiver(
+            rabbitTemplate, caseRepository, uacQidLinkRepository, fieldworkFollowupBuilder);
+
+    UUID testCaseId = UUID.randomUUID();
+    UacQidLink uacQidLink = new UacQidLink();
+    uacQidLink.setCaseId(testCaseId.toString());
+    Case caze = new Case();
+    caze.setFieldCoordinatorId("Noodlebiscuits");
+    FieldworkFollowup fieldworkFollowup = new FieldworkFollowup();
+
+    // Given
+    when(uacQidLinkRepository.findByQid(eq("testQid"))).thenReturn(Optional.of(uacQidLink));
+    when(caseRepository.findByCaseId(eq(testCaseId))).thenReturn(Optional.of(caze));
+    when(fieldworkFollowupBuilder.buildFieldworkFollowup(eq(caze), eq("dummy"), eq("dummy")))
```

Comment:
general comment, I don't think matchers are required here, they're only required when one of the arguments is a matcher.  Doesn't need fixing, just less typing in the future.